### DISCUSSION TITLE
Extract value from json field when using custom Content-Type

### DISCUSSION THREAD

**Discussion by kierangirvan** [mockoon/mockoon#792] [Help] [answered]
Hi,

The application I'm trying to mock is using a custom Content-Type, albeit still in json form, but the Content-Type being passed to the mock (and real application) doesn't conform to application/json

An example POST request body:
`{
  "storeId": 2,
  "items": [
    {
      "productId": "5057006847998",
      "quantity": 1,
      "measurement": "EA"
    }
  ]
}` 

Now I want to be able to capture the productId from the request as this is needed in the response.

Is there a way using a regex to extract this value?  I know by switching over to a supported Content-Type such as application/json that I can use `{{body 'items.0.productId'}}` but this isn't possible due to the customer Content-Type at play.

I've tried using the stringify helper but this didn't work. I've also tried to replicate the example given in the documentation [HERE](https://mockoon.com/docs/latest/templating/mockoon-helpers/#stringify) but that didn't work either.

Really appreciate any help with this.

Many thanks,

**Comment by Gogolian** [comment]
Did you maybe try with substr and indexof ?
https://mockoon.com/docs/latest/templating/mockoon-helpers/#substr
https://mockoon.com/docs/latest/templating/mockoon-helpers/#indexof

**Reply by kierangirvan** [reply]
Thanks @Gogolian 

I've managed to extract the value from the request using a combination of substr and indexOf.  The small concern I have is the length of the productId is not always the same, so I've had to also find the index of a latter part of the request body (in this case the quantity), if in the future the request body itself changes then this is prone to failure.

So ideally looking for a way to parse the request body as json, and point to the desired path.  But if this isn't an option then your suggestion will work for now at least.

**Reply by 255kb** [reply]
The only other approach I see would be to create another Mockoon environment used as a proxy to your existing environment. It's sole purpose would be to modify the Content-Type helper in the proxy request headers: 

![image](https://user-images.githubusercontent.com/7489814/184352123-7f763b16-6482-4b7b-9d0e-464195a698da.png)

Also, headers support the same templating syntax, so you can use helpers like `if` or anything else to conditionally set your content-type header.

Here is the JSON for the env if you want to test: 

```JSON
{
    "uuid": "1948a1f7-7dd5-4b47-a392-b46e65f2bfc6",
    "lastMigration": 22,
    "name": "Proxy",
    "endpointPrefix": "",
    "latency": 0,
    "port": 3001,
    "hostname": "0.0.0.0",
    "routes": [],
    "proxyMode": true,
    "proxyHost": "http://localhost:3002",
    "proxyRemovePrefix": false,
    "tlsOptions": {
        "enabled": false,
        "type": "CERT",
        "pfxPath": "",
        "certPath": "",
        "keyPath": "",
        "caPath": "",
        "passphrase": ""
    },
    "cors": true,
    "headers": [],
    "proxyReqHeaders": [
        {
            "key": "Content-Type",
            "value": "application/json"
        }
    ],
    "proxyResHeaders": []
}
```

**Reply by kierangirvan** [reply]
Thank you so much @255kb 

I'll try the proxy route and come back to you.

**Reply by kierangirvan** [reply]
Pleased to confirm this is now working by using the proxy, and adjusting the content-type allowing me to use the body helper.

**Comment by kierangirvan** [comment, marked as answer]
@255kb suggestion (above) of using the proxy feature of mockoon to adjust the content-type allowed me to make use of the body helper, so I was able to fetch the desired value from the request and reuse in the payload.

**Reply by 255kb** [reply]
Glad it worked!
